File: Android/src/main/java/com/thinkincode/quran/sdk/database/HifdhTips.java
package com.thinkincode.quran.sdk.database;

import android.content.res.Resources;
import android.support.annotation.NonNull;

import com.thinkincode.quran.sdk.R;

import java.util.Random;

/**
 * Helper class which provides Hifdh tips.
 */
public class HifdhTips {

    @NonNull
    private final Resources resources;

    @NonNull
    private final Random random;

    /**
     * Constructor.
     *
     * @param resources the application's {@link Resources} instance.
     */
    public HifdhTips(@NonNull Resources resources) {
        this.resources = resources;
        this.random = new Random();
    }

    /**
     * @return a randomly selected Hifdh tip.
     */
    @NonNull
    public String getNextRandom() {
        int index = random.nextInt(getSize());
        return getTip(index);
    }

    String getTip(int index) {
        return getTipsArray()[index];
    }

    int getSize() {
        return getTipsArray().length;
    }

    private String[] getTipsArray() {
        return resources.getStringArray(R.array.hifdh_tips);
    }
}

File: Android/src/main/java/com/tazkiyatech/quran/sdk/database/HifdhTips.kt
package com.tazkiyatech.quran.sdk.database

import android.content.res.Resources
import com.tazkiyatech.quran.sdk.R
import java.util.*

/**
 * Helper class which provides Hifdh tips.
 *
 * @property resources The application's [Resources] instance.
 */
class HifdhTips(private val resources: Resources) {

    private val random: Random = Random()

    /**
     * @return a randomly selected Hifdh tip.
     */
    val nextRandom: String
        get() = getTip(random.nextInt(size))

    /**
     * The number of Hifdh tips available.
     */
    val size: Int
        get() = tipsArray.size

    /**
     * The [Array] of Hifdh tips.
     */
    private val tipsArray: Array<String>
        get() = resources.getStringArray(R.array.hifdh_tips)

    /**
     * @param index the index of the Hifdh tip to get.
     * @return the Hifdh tip at the specified index.
     */
    fun getTip(index: Int): String {
        return tipsArray[index]
    }
}
